_playwright-tests/UI/RepositoryPackages.spec.ts

```diff
@@ -0,0 +1,95 @@
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { closePopupsIfExist, getRowByNameOrUrl } from './helpers/helpers';
+export const randomName = () => (Math.random() + 1).toString(36).substring(2, 6);
+export const repoName = `${randomName()}`;
+export const edited_repo = `${repoName}-Edited`;
+export const rank = () => Math.floor(Math.random() * 10 + 1).toString();
+export const randomNum = () =>
+  Math.floor(Math.random() * 10 + 1)
+    .toString()
+    .padStart(2, '0');
+
+test.describe('Snapshot Package Count and List', async () => {
+  test('Verify package count and search functionality in snapshot details', async ({ page }) => {
+    await navigateToRepositories(page);
+    await closePopupsIfExist(page);
+    await deleteAllRepos(page);
+
+    await test.step('Create a repository', async () => {
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('Snapshotting').click();
+      await page.getByLabel('URL').fill('https://jlsherrill.fedorapeople.org/fake-repos/signed/');
+      await page.getByRole('button', { name: 'Save', exact: true }).click();
+    });
+
+    await test.step('Wait for status to be "Valid"', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+
+    await test.step('Verify the package count matches the snapshot', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      const packageCountValue = await row.getByTestId('package_count_button').textContent();
+      await page.getByRole('button', { name: 'Kebab toggle' }).click();
+      await page.getByRole('menuitem', { name: 'View all snapshots' }).click();
+      // click on the first row to view the snapshot details
+      const snapshotPackagesColumn = await page
+        .getByTestId('snapshot_package_count_button')
+        .textContent();
+      // assert the package count matches the after snapshot packages count
+      expect(packageCountValue).toBe(snapshotPackagesColumn);
+      await page.getByText('Close').click();
+    });
+
+    // Edit the repository to change number of packages
+    await test.step('Update the repository', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      await row.getByLabel('Kebab toggle').click();
+      await row.getByRole('menuitem', { name: 'Edit' }).click();
+      await page.getByPlaceholder('Enter name', { exact: true }).fill(edited_repo);
+      await page
+        .getByLabel('URL')
+        .fill('http://jlsherrill.fedorapeople.org/fake-repos/needed-errata/');
+      await page.getByRole('button', { name: 'Save changes', exact: true }).click();
+      const edited_row = await getRowByNameOrUrl(page, edited_repo);
```

Comment:
nitpick: in typescript the norm is to use `camelCase` for variable names and not `snake_case`, so this should be `editedRow` 💅🏼😅 (same goes for a lot of other variables in this file)

Reply:
Fixed!